### ISSUE TITLE
Particle family mappings of AHF haloes

### ISSUE THREAD

**Issue by Martin-Rey** [pynbody/pynbody#691]
Hi @apontzen and @cphyc 

I am having some interesting behaviours comparing my simulated data in a Ramses and Tipsy formats, both with the same AHF catalogue. When loading the snapshot as Tipsy, the particle families of the halo are well setup and match the AHF record
```
In [27]: type(s)
Out[27]: pynbody.snapshot.tipsy.TipsySnap

In [23]: s.halos()[1].properties['npart']
Out[23]: 7332330

In [24]: s.halos()[1]
Out[24]: <SimSnap "./output_00113_tipsy/output_00113_fullbox.tipsy:halo_1" len=7332330>

In [19]: s.halos()[1].properties['n_gas']
Out[19]: 3608832

In [25]: s.halos()[1].g
Out[25]: <SimSnap "./output_00113_tipsy/output_00113_fullbox.tipsy:halo_1::gas" len=3608832>

In [18]: s.halos()[1].properties['n_star']
Out[18]: 3955

In [26]: s.halos()[1].st
Out[26]: <SimSnap "./output_00113_tipsy/output_00113_fullbox.tipsy:halo_1::star" len=3955>

In [39]: s.halos()[1].dm
Out[39]: <SimSnap "./output_00113_tipsy/output_00113_fullbox.tipsy:halo_1::dm" len=3719543>
```

However, when I repeat the same operation sloading my simulation data as a Ramses snapshot, the stars are wrongly mapped onto the DM family, yielding 0 stars associated with the halo and too many DM particles (which I guess is inferred as the difference between `npart`, `n_stars` and `n_gas`).

```

In [29]: type(s)
Out[29]: pynbody.snapshot.ramses.RamsesSnap

In [30]: s.halos()[1].properties['n_star']
Out[30]: 3955

In [31]: s.halos()[1].st
Out[31]: <SimSnap "./output_00113:halo_1::star" len=0>

In [32]: s.halos()[1].properties['n_gas']
Out[32]: 3608832

In [34]: s.halos()[1].g
Out[34]: <SimSnap "./output_00113:halo_1::gas" len=3608832>

In [35]: s.halos()[1].properties['npart']
Out[35]: 7332330

In [36]: s.halos()[1]
Out[36]: <SimSnap "./output_00113:halo_1" len=7332330>

In [37]: s.halos()[1].dm
Out[37]: <SimSnap "./output_00113:halo_1::dm" len=3723498>

```

I am unsure what could be causing this, and I only see this issue for the newer formats of Ramses with family and particle tags. This makes me think that the `particle distinguisher` logic is not well setup, whether this is a config or an actual bug. Could you point me to the code in the AHF loader that assigns halo particles to each family to help me track this down?

Thanks,
Martin

**Comment by Martin-Rey** [comment]
I have confirmed that this issue arises only on my runs with the newer ramses format, which also contain additional particle families in the form of gas and stellar tracers.

It seems to me that pynbody fails to remap the AHF ids onto the indices of particles of the right families. This is currently implemented using a lot of acrobatics and format-specific logic:
https://github.com/pynbody/pynbody/blob/aa4bd2a0e9fb411d68c246daac7a2752e0020d2f/pynbody/halo/ahf.py#L326-L348

@cphyc The `pass` for Ramses snapshots was introduced recently in #661. What was the logic for it and was it ever tested on the newer format with/without tracers?

**Comment by Martin-Rey** [comment]
Fixed by #697
